File: controlePedidos/src/main/java/com/br/chocolatePontoVirgula/model/entity/Pedido.java
package com.br.chocolatePontoVirgula.model.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.sql.Timestamp;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Pedido {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    protected int codigo;

    protected Timestamp dataPedido;
    protected int codigoCliente;
    protected String enderecoEntrega;
    protected boolean situacao;
    protected double valorTotal;
    protected int quantidadeTotal;
    protected int percentualDesconto;



    public boolean isFechado() {
        return this.situacao;
    }

    public void aplicarDesconto() {
        if (!isFechado()) {
            valorTotal *= (1 - (this.percentualDesconto * 100));
        } else {
            System.out.println("Não foi possível aplicar o desconto");
        }
    }
}

File: controlePedidosFront/src/app/shared/models/pedido.ts
export class Pedido {
    id: number;
    cliente: string;
    enderecoEntrega: string;
    aberto: boolean;
    subtotal: number;
    desconto: number;
    total: number;

    constructor(codigo: number,
        cliente: string,
        enderecoEntrega: string,
        situacao: boolean,
        subtotal: number,
        desconto: number,
        total: number){
            this.id = codigo;
            this.cliente = cliente;
            this.enderecoEntrega= enderecoEntrega;
            this.aberto =situacao;
            this.subtotal = subtotal;
            this.desconto = desconto;
            this.total = total;
        }
        
      }
  
  